File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03502/p03502_num6_parsed.py
N = int(input())

def is_harshad_number(n):
    sum_of_digits = sum([int(digit) for digit in str(n)])
    return n % sum_of_digits == 0

if is_harshad_number(N):
    print("Yes")
else:
    print("No")

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03502/p03502_num6_parsed.cpp
#include <iostream>
using namespace std;

int main() {
    int N;
    cin >> N;

    int sum = 0;
    int num = N;
    while (num > 0) {
        sum += num % 10;
        num /= 10;
    }

    if (N % sum == 0) {
        cout << "Yes" << endl;
    } else {
        cout << "No" << endl;
    }

    return 0;
}